components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/ColorProcessor.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.processor
+
+import android.content.Context
+import android.graphics.Color.rgb
+import androidx.palette.graphics.Palette
+import mozilla.components.browser.icons.Icon
+import mozilla.components.browser.icons.IconRequest
+
+/**
+ * [IconProcessor] implementation to extract the dominant color from the icon.
+ */
+class ColorProcessor : IconProcessor {
+
+    override fun process(context: Context, request: IconRequest, resource: IconRequest.Resource?, icon: Icon): Icon {
+        if (icon.bitmap == null || icon.color != null) return icon
+
+        val dominantColor = Palette.from(icon.bitmap).generate().getDominantColor(DEFAULT_COLOR)
```

Comment:
A default color worked fine when this was app code. Now as a component I think we should just return no color so that the app can pick a good default for its use case.

Reply:
Sounds good, I'll update it so that the icon isn't modified if no colour is found.